python/tests/test_nvcomp_codec.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) 2023, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+import json
+
+import numcodecs
+import numpy as np
+import pytest
+import zarr
+from numpy.testing import assert_equal
+
+from kvikio.nvcomp_codec import NvCompBatchCodec
+
+NVCOMP_CODEC_ID = "nvcomp_batch"
+
+LZ4_ALGO = "LZ4"
+GDEFLATE_ALGO = "Gdeflate"
+SNAPPY_ALGO = "snappy"
+ZSTD_ALGO = "zstd"
+
+SUPPORTED_CODECS = [LZ4_ALGO, GDEFLATE_ALGO, SNAPPY_ALGO, ZSTD_ALGO]
+
+
+def _get_codec(algo: str):
+    codec_args = {"id": NVCOMP_CODEC_ID, "algorithm": algo}
+    return numcodecs.registry.get_codec(codec_args)
+
+
+@pytest.mark.parametrize("algo", SUPPORTED_CODECS)
+def test_codec_registry(algo: str):
+    codec = _get_codec(algo)
+    assert isinstance(codec, numcodecs.abc.Codec)
+
+
+@pytest.mark.parametrize("algo", SUPPORTED_CODECS)
+def test_basic(algo: str):
+    shape = (16, 16)
+    codec = NvCompBatchCodec(algo)
+
+    # Create data.
+    dtype = np.float32
+    data = np.ones(shape, dtype=dtype)
+    # Do roundtrip.
+    comp_data = codec.encode(data)
+    # Decompress and cast to original data type/shape.
+    decomp_data = codec.decode(comp_data).view(dtype).reshape(shape)
+
+    assert_equal(decomp_data, data)
+
+
+@pytest.mark.parametrize("algo", SUPPORTED_CODECS)
+def test_basic_zarr(algo: str):
+    shape = (16, 16)
+    chunks = (8, 8)
```

Comment:
Would it make sense to have a `@pytest.mark.parametrize()` or a fixture to test a range of shapes and chunks in most of the tests?

Reply:
* Added comments.
* Updated tests with parameterized fixtures to test various combinations of inputs/chunks.